### PR TITLE
US-391911-1:changes to support authentication in hz c/s mode in k8s

### PR DESCRIPTION
**Pull request by SushmaReddyLoka**
Changes to support hz c/s authentication. We take username and password, and create secrets out of this and mount volumes created out of these secrets on container files. Tomcat gets these values through prweb.xml only when hz c/s mode is enabled.

### CODE REVIEW

File: scripts/docker-entrypoint.sh [hunk 1]
```diff
@@ -126,6 +129,25 @@ else
...
+   export SECRET_HZ_CS_AUTH_USERNAME=${HZ_CS_AUTH_USERNAME}
+fi
+
+if [ -e "$hazelcast_password_file" ]; then
+   export SECRET_HZ_CS_AUTH_PASSWORD=$(<${hazelcast_password_file})
+else
+   export SECRET_HZ_CS_AUTH_PASSWORD=${HZ_CS_AUTH_PASSWORD}
+fi
+
+if [ "$CLIENT_MODE" = true ]; then
```
Comment (Saurabh-16) [inline scripts/docker-entrypoint.sh:144]: Will it be `CLIENT_MODE` or `HZ_CLIENT_MODE`?

Comment (SushmaReddyLoka) [reply scripts/docker-entrypoint.sh:144]: Yes, it should be HZ_CLIENT_MODE.

File: scripts/docker-entrypoint.sh [hunk 2]
```diff
@@ -126,6 +129,25 @@ else
...
+   export SECRET_HZ_CS_AUTH_USERNAME=${HZ_CS_AUTH_USERNAME}
+fi
+
+if [ -e "$hazelcast_password_file" ]; then
+   export SECRET_HZ_CS_AUTH_PASSWORD=$(<${hazelcast_password_file})
+else
+   export SECRET_HZ_CS_AUTH_PASSWORD=${HZ_CS_AUTH_PASSWORD}
+fi
+
+if [ "HZ_CLIENT_MODE" = true ]; then
```
Comment (Saurabh-16) [inline scripts/docker-entrypoint.sh:144]: Don't we need `==` here

**Comment by Saurabh-16** [comment]
Please update the repo readme with these 2 new environment variables in Dockerfile under hazelcast settings.

File: scripts/docker-entrypoint.sh (hunk 2 above)
Comment (SushmaReddyLoka) [reply scripts/docker-entrypoint.sh]: I think both of them will work, we tested this with the variable with false and true values  and it worked well, but to be consistent changed it to "==" and pushed changes.

**Comment by SushmaReddyLoka** [comment]
> Please update the repo readme with these 2 new environment variables in Dockerfile under hazelcast settings.

done

**Review by MadhuriArugula — APPROVED** [review]

**Review by kishorv10 — APPROVED** [review]

File: scripts/docker-entrypoint.sh [hunk 3]
```diff
@@ -126,6 +129,25 @@ else
...
+   export SECRET_HZ_CS_AUTH_PASSWORD=${HZ_CS_AUTH_PASSWORD}
+fi
+
+if [ "HZ_CLIENT_MODE" == true ]; then
+    if [ "$SECRET_HZ_CS_AUTH_USERNAME" == "" ] || [ "$SECRET_HZ_CS_AUTH_PASSWORD" == "" ]; then
+        echo "HZ_CS_AUTH_USERNAME & HZ_CS_AUTH_PASSWORD must be specified in hazelcast client server mode deployments.";
+        exit 1
+    fi
+fi
+
```
Comment (MadhuriArugula) [inline scripts/docker-entrypoint.sh:150]: Can you add test cases to the changes made around entrypoint. Please refer https://github.com/pegasystems/docker-pega-web-ready/blob/master/tests/pega-web-ready-testcases.yaml to write tests for environment variables and validating mounted secrets.

**Review by MadhuriArugula — CHANGES_REQUESTED** [review]

File: scripts/docker-entrypoint.sh (hunk 3 above)
Comment (PraneethaYanala) [reply scripts/docker-entrypoint.sh:150]: @MadhuriArugula added the test cases

**Review by Saurabh-16 — APPROVED** [review]

**Review by MadhuriArugula — APPROVED** [review]

**Pull request merged**